### PR TITLE
feat(DTFS2-6628): add read only limitation on UI elements in gef-ui

### PR DESCRIPTION
**Pull request by lmarrai-sw**
## Introduction

As part of adding the read-only role, we need to check that the UI components in `gef-ui` will behave correctly for users with the read-only role.

## Resolution

We have:
1. updated the `gef-ui` primary navigation component to show only the Dashboard link (not Reports or Users)
2. changed `application-details/automatic-cover.njk` to hide the "Check if you're eligible for automatic cover" link if page is in preview mode (not just for admins)
3. fixed a bug on `application-details/automatic-cover.njk` page that showed the "Check if you're eligible for automatic cover" link to admin if the eligibility status code was "In progress"
4. changed `applicationReferenceAndActions.njk` to only show the abandon, edit ref name, and edit internal ref name links for makers - previously any non-admins could see these (and moved the logic for this from `applicationReferenceAndActions.njk` to the controller)
5. changed the link for updating unissued facilities to only show for makers - previously any non-admins could see this (and made the logic for this clearer)

## Misc

Other small improvements:
- we deleted the `accessibility-statement.njk` file in `gef-ui` as it was not used.
- we did a small refactor to `footer.njk` in `gef-ui`
- we fixed a minor bug in `getUserAuthorisationLevelsToApplication` that caused a console error to be logged if it was called for a user that had a role other than `maker` or `checker` (it gets called on a lot of pages, including whenever users view gef applications)
- we moved `canUpdateUnissuedFacilitiesCheck ` out from a generic `helpers.js` file to its own file next to the only file that uses it

### CODE REVIEW

File: gef-ui/server/controllers/unissued-facilities/index.js [hunk 1]
```diff
@@ -116,6 +116,8 @@ const changeIssuedToUnissuedFacility = async (req, res) => {
     const { details } = await api.getFacility(facilityId);
     const hasBeenIssued = JSON.stringify(details.hasBeenIssued);
 
+    // TODO DTFS2-6628: template (and any sub-templates) will need to be checked if we
+    // decide not to validate the route for this function in DTFS2-6627
     return res.render('partials/issued-facility-to-unissued.njk', {
```
Comment (lmarrai-sw) [inline gef-ui/server/controllers/unissued-facilities/index.js:121]: Flagging TODO. This depends on what happens in #2056

File: gef-ui/server/controllers/unissued-facilities/index.js [hunk 2]
```diff
@@ -327,6 +329,8 @@ const postChangeIssuedToUnissuedFacility = async (req, res) => {
       errMsg: `Select if your bank has already issued this ${facilityTypeString} facility`,
     });
 
+    // TODO DTFS2-6628: template (and any sub-templates) will need to be checked if we
+    // decide not to validate the route for this function in DTFS2-6627
     return res.render('partials/issued-facility-to-unissued.njk', {
```
Comment (lmarrai-sw) [inline gef-ui/server/controllers/unissued-facilities/index.js:334]: Flagging TODO. This depends on what happens in #2056

File: gef-ui/server/models/application.js [hunk 3]
```diff
@@ -94,6 +94,7 @@ class Application {
     try {
       const application = await getApplication(id);
 
+      // TODO DTFS2-6628: this doesn't look right - what about if the user has access to all banks?
       if (application.bank.id !== user.bank.id && !user.roles.includes(ADMIN)) {
         return null;
       }
```
Comment (lmarrai-sw) [inline gef-ui/server/models/application.js:100]: @abhi-markan would be curious to get your thoughts here. This logic seems to say that you can view the gef application if either:
- your bank matches the application's bank, or
- you are an admin

but I thought it should be either:
- your bank matches the application's bank, or
- your bank is the special "All" bank (regardless of admin or not)

Do you agree with me or do you think I have misunderstood?

File: gef-ui/component-tests/includes/application-details/applicationReferenceAndActions.component-test.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    it('should not render when the page is rendered without the userRoles param', () => {
+      wrapper = render({
+        dealId,
+        status,
+      });
+      wrapper.expectLink(cloneButtonSelector).notToExist();
+    });
+
+    it('should not render when the page is rendered without the userRoles param', () => {
```
[the remark this answers is not in the document]
Comment (lmarrai-sw) [reply gef-ui/component-tests/includes/application-details/applicationReferenceAndActions.component-test.js]: good spot!

File: gef-ui/templates/partials/accessibility-statement.njk [hunk 5]
```diff
@@ -1,65 +0,0 @@
-{% extends "index.njk" %}
```
[the remark this answers is not in the document]
Comment (lmarrai-sw) [reply gef-ui/templates/partials/accessibility-statement.njk:1]: `gef-ui` doesn't actually use it anywhere - I couldn't find any routes in gef-ui that rendered this template

There's an identically named template in `portal` that is used

File: gef-ui/server/controllers/application-details/index.js [hunk 6]
```diff
@@ -124,11 +126,10 @@ function buildBody(app, previewMode, user) {
     isUkefReviewPositive: ukefReviewPositive,
     ukefDecisionAccepted: hasUkefDecisionAccepted,
     coverDatesConfirmed: coverDates,
-    renderReviewDecisionLink: ukefReviewAvailable && ukefReviewPositive && !coverDates && !hasUkefDecisionAccepted && !app.userRoles.includes(ADMIN),
+    renderReviewDecisionLink: ukefReviewAvailable && ukefReviewPositive && !coverDates && !hasUkefDecisionAccepted && app.userRoles.includes(MAKER),
```
[the remark this answers is not in the document]
Comment (lmarrai-sw) [reply gef-ui/server/controllers/application-details/index.js:129]: The `GET /application-details/:dealId/review-decision' route that is linked to when this is `true` has the `validateRole(MAKER)` middleware, so only makers can access the route currently :)

File: gef-ui/component-tests/includes/footer.component-test.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    beforeEach(() => {
+      wrapper = render({
+        user: {
+          roles: [nonMakerOrCheckerRole],
+        },
+      });
+    });
+
+    it('should NOT render the feedback link', () => {
+      wrapper.expectElement('[data-cy="feedback-link"]').notToExist();
```
[the remark this answers is not in the document]
Comment (lmarrai-sw) [reply gef-ui/component-tests/includes/footer.component-test.js:49]: It's a good question - this is the same in `portal` as well, did we spot this in DTFS2-6624 at all?

@abhi-markan should read-only and admin users be able to view the feedback link in the footer?

[the remark this answers is not in the document]
Comment (AlexBramhill) [reply gef-ui/component-tests/includes/footer.component-test.js:49]: Ah in which case I've caught it in GEF-UI and not in portal 😄 Should be able to give feedback

File: gef-ui/server/controllers/unissued-facilities/index.js (hunk 1 above)
Comment (lmarrai-sw) [reply gef-ui/server/controllers/unissued-facilities/index.js]: We added the role validation in #2056 so there's nothing to do here

File: gef-ui/component-tests/includes/footer.component-test.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (lmarrai-sw) [reply gef-ui/component-tests/includes/footer.component-test.js:49]: This is going to be changed in #2059

File: gef-ui/server/models/application.js (hunk 3 above)
Comment (lmarrai-sw) [reply gef-ui/server/models/application.js:100]: This is outside the scope of the current ticket so I'm going to close this off and will investigate it separately

**Review by AlexBramhill — APPROVED** [review]

**Pull request merged**